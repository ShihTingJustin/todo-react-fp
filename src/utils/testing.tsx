import React, { FC, ReactElement } from 'react';
import { Provider } from 'react-redux';

// import {  LocationProvider } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';

import { store } from '@Redux/store';

// const history = createHistory(window);
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <Provider store={store}>{children}</Provider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render, AllTheProviders };
