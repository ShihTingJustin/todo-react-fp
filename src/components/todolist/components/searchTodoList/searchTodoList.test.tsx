import renderer from 'react-test-renderer';
import { AllTheProviders, render, screen } from '@Utils/testing';
import SearchTodoList from './searchTodoList';

describe('Test Button component', () => {
  const props = {
    result: {
      id: '63a99ee9e6e183933c167649',
      title: 'Grocery List',
      isError: false,
      isLoading: false,
      todo: [
        {
          id: '63a9b83550511904a6e0fe81',
          listId: '63a99ee9e6e183933c167649',
          completed: false,
          title: 'qwe',
        },
      ],
    },
  };
  const component = AllTheProviders({ children:<SearchTodoList {...props} /> });

  it('[Render] match snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('[Render] toBeInTheDocument', () => {
    render(component);
    const element = screen.getByTestId('searchTodoList');
    expect(element).toBeInTheDocument();
  });
});
