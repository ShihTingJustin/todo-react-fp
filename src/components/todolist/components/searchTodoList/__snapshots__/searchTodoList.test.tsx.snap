// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component [Render] match snapshot 1`] = `
<div
  data-testid="searchTodoList"
>
  <div
    className="flex justify-between items-center pl-6 pr-5 h-[3.125rem]"
    data-testid="todoListHeader"
  >
    <div
      className="text-title-2 text-primary-blue h-[2.625rem]"
    >
      Grocery List
    </div>
    <div
      className="right-wrap"
    >
      <button
        className="ant-btn ant-btn-circle ant-btn-text ant-btn-sm ant-btn-icon-only mr-6"
        data-disabled={false}
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="rm-new-reminder-icon glyph-box"
          fill="#0071EB"
          version="1.1"
          viewBox="0 0 80.567626953125 80.6142578125"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="matrix(1 0 0 1 -12.4511572265626 75.537109375)"
          >
            <path
              d="M12.4512-35.2539C12.4512-32.8613 14.4531-30.9082 16.7969-30.9082L48.3887-30.9082L48.3887 0.683594C48.3887 3.02734 50.3418 5.0293 52.7344 5.0293C55.127 5.0293 57.1289 3.02734 57.1289 0.683594L57.1289-30.9082L88.6719-30.9082C91.0156-30.9082 93.0176-32.8613 93.0176-35.2539C93.0176-37.6465 91.0156-39.6484 88.6719-39.6484L57.1289-39.6484L57.1289-71.1914C57.1289-73.5352 55.127-75.5371 52.7344-75.5371C50.3418-75.5371 48.3887-73.5352 48.3887-71.1914L48.3887-39.6484L16.7969-39.6484C14.4531-39.6484 12.4512-37.6465 12.4512-35.2539Z"
            />
          </g>
        </svg>
      </button>
      <button
        className="ant-btn ant-btn-circle ant-btn-text ant-btn-sm ant-btn-icon-only"
        data-disabled={false}
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="list-detail-icon glyph-box"
          fill="#0071EB"
          version="1.1"
          viewBox="0 0 99.6099853515625 99.6572265625"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="matrix(1 0 0 1 -8.740039062500045 85.05859375)"
          >
            <path
              d="M58.5449 14.5508C85.791 14.5508 108.35-8.05664 108.35-35.2539C108.35-62.5 85.7422-85.0586 58.4961-85.0586C31.2988-85.0586 8.74023-62.5 8.74023-35.2539C8.74023-8.05664 31.3477 14.5508 58.5449 14.5508ZM58.5449 6.25C35.498 6.25 17.0898-12.207 17.0898-35.2539C17.0898-58.3008 35.4492-76.7578 58.4961-76.7578C81.543-76.7578 100-58.3008 100.049-35.2539C100.098-12.207 81.5918 6.25 58.5449 6.25ZM35.7422-28.1738C39.6484-28.1738 42.8223-31.3477 42.8223-35.3027C42.8223-39.209 39.5996-42.3828 35.7422-42.3828C31.7871-42.3828 28.6133-39.209 28.6133-35.3027C28.6133-31.3477 31.7871-28.1738 35.7422-28.1738ZM58.4961-28.1738C62.4023-28.1738 65.625-31.3477 65.625-35.3027C65.625-39.209 62.4023-42.3828 58.4961-42.3828C54.5898-42.3828 51.3672-39.209 51.3672-35.3027C51.3672-31.3477 54.5898-28.1738 58.4961-28.1738ZM81.2988-28.1738C85.2051-28.1738 88.3789-31.3477 88.3789-35.3027C88.3789-39.209 85.2051-42.3828 81.2988-42.3828C77.3438-42.3828 74.1699-39.209 74.1699-35.3027C74.1699-31.3477 77.3438-28.1738 81.2988-28.1738Z"
            />
          </g>
        </svg>
      </button>
    </div>
  </div>
  <div
    className="flex flex-col grow"
  >
    <div
      className="\\"rowgroup\\" false"
      data-testid="todoItem"
      tabIndex={0}
    >
      <div
        className="content flex items-center py-3 pr-5 pl-6 w-full"
      >
        <div
          className="mr-3"
        >
          <input
            checked={false}
            className="ant-input toggle hidden"
            id="63a9b83550511904a6e0fe81"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="checkbox"
            value=""
          />
          <label
            htmlFor="63a9b83550511904a6e0fe81"
            onClick={[Function]}
          >
            <div
              className="w-[1.375rem] h-[1.375rem]"
            >
              <svg
                className="glyph-box"
                fill="#c4c4c6"
                role="presentation"
                version="1.1"
                viewBox="0 0 99.6099853515625 99.6572265625"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  transform="matrix(1 0 0 1 -8.740039062500045 85.05859375)"
                >
                  <path
                    d="M58.5449 14.5508C85.791 14.5508 108.35-8.05664 108.35-35.2539C108.35-62.5 85.7422-85.0586 58.4961-85.0586C31.2988-85.0586 8.74023-62.5 8.74023-35.2539C8.74023-8.05664 31.3477 14.5508 58.5449 14.5508ZM58.5449 6.25C35.498 6.25 17.0898-12.207 17.0898-35.2539C17.0898-58.3008 35.4492-76.7578 58.4961-76.7578C81.543-76.7578 100-58.3008 100.049-35.2539C100.098-12.207 81.5918 6.25 58.5449 6.25Z"
                  />
                </g>
              </svg>
            </div>
          </label>
        </div>
        <div
          className="todo-item w-full"
        >
          <div
            className="flex items-center w-full border-b border-border-gray2"
          >
            <input
              className="ant-input ant-input-borderless edit w-full text-content-1"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value="qwe"
            />
            <div
              className="delete-button px-3 py-1 cursor-pointer"
              onClick={[Function]}
            >
              <span
                aria-label="close"
                className="anticon anticon-close"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
